webapp/components/signup/components/signup_ldap.jsx

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import FormError from 'components/form_error.jsx';
+
+import * as GlobalActions from 'actions/global_actions.jsx';
+import {track} from 'actions/analytics_actions.jsx';
+
+import * as Utils from 'utils/utils.jsx';
+import Client from 'client/web_client.jsx';
+
+import React from 'react';
+import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
+import {browserHistory, Link} from 'react-router/es6';
+
+import logoImage from 'images/logo.png';
+
+export default class SignupLdap extends React.Component {
+    static get propTypes() {
+        return {
+            location: React.PropTypes.object
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.handleLdapSignup = this.handleLdapSignup.bind(this);
+
+        this.handleLdapIdChange = this.handleLdapIdChange.bind(this);
+        this.handleLdapPasswordChange = this.handleLdapPasswordChange.bind(this);
+
+        this.state = ({
+            ldapError: ''
+        });
+    }
+
+    handleLdapSignup(e) {
+        e.preventDefault();
+
+        this.setState({ldapError: ''});
+
+        Client.webLoginByLdap(
+            this.state.ldapId,
+            this.state.ldapPassword,
+            null,
+            () => {
+                if (this.props.location.query.id || this.props.location.query.h) {
+                    Client.addUserToTeamFromInvite(
+                        this.props.location.query.d,
+                        this.props.location.query.h,
+                        this.props.location.query.id,
+                        () => {
+                            this.finishSignup();
+                        },
+                        () => {
+                            // there's not really a good way to deal with this, so just let the user log in like normal
+                            this.finishSignup();
+                        }
+                    );
+
+                    return;
+                }
+
+                this.finishSignup();
+            },
+            (err) => {
+                if (err.id === 'ent.ldap.do_login.user_not_registered.app_error' || err.id === 'ent.ldap.do_login.user_filtered.app_error') {
```

Comment:
I think this is left over from a combination of the login revamps (we didn't want to show the "LDAP user not registered" error when there was no account found) and from when we were obfuscating the different login error messages. It should be safe to remove both of them